### ISSUE TITLE
pytorch2onnx with wrong axis for Flatten op

### ISSUE THREAD

**Issue by RunningLeon** [pytorch/pytorch#54473]
## Description
Hi, I try to create tensorrt engine from pytorch->onnx->tensorrt. There is an error with `Flatten`.
From onnx graph, you can see `Flatten_31` op flattens tensor `scores` of shape [2,2,10] with attribute `axis=3`.  Since `ndim` is 3 for `scores` and thus leads to this asseration error. Thus, should `axis` be 2 if it is starting from 0?
Could anyone help me with this issue? 

## sample code to convert a onnx
```
import torch
import onnx

class WrapFunction(torch.nn.Module):

    def __init__(self, wrapped_function):
        super(WrapFunction, self).__init__()
        self.wrapped_function = wrapped_function

    def forward(self, *args, **kwargs):
        return self.wrapped_function(*args, **kwargs)
    
def test_flatten(boxes, scores, selected_indices):
    batch_inds, cls_inds = selected_indices[:, 0], selected_indices[:, 1]
    box_inds = selected_indices[:, 2]
    boxes = boxes[batch_inds, box_inds, :]
    scores = scores[batch_inds, cls_inds, box_inds]
    dets = torch.cat([boxes, scores[:, None]], dim=1)
    return dets, batch_inds, cls_inds

# create input data
batch_size = 2
num_box = 10
num_class = 2
num_det = 5
boxes = torch.rand(batch_size, num_box, 4) * 10
scores = torch.rand(batch_size, num_class, num_box) * 10
batch_inds = torch.randint(batch_size, (num_box, ))
cls_inds = torch.randint(num_class, (num_box, ))
box_inds = torch.randint(num_box, (num_box, ))
selected_indices = torch.cat([batch_inds[:, None], cls_inds[:, None], box_inds[:, None]], dim=1)
input_data = (boxes, scores, selected_indices)

wrapped_model = WrapFunction(test_flatten)
wrapped_model.cpu().eval()
onnx_file = 'test_flatten.onnx'
with torch.no_grad():
    torch.onnx.export(
        wrapped_model, 
        input_data,
        onnx_file,
        export_params=True,
        keep_initializers_as_inputs=True,
        input_names=['boxes', 'scores', 'selected_indices'],
        output_names=['dets', 'batch_indices', 'labels'],
        opset_version=11)
onnx_model = onnx.load(onnx_file)

# run with onnxruntime
sess = ort.InferenceSession(onnx_file)
onnx_outputs = sess.run(None, {
    'scores': scores.detach().numpy(),
    'boxes': boxes.detach().numpy(),
    'selected_indices':selected_indices.detach().numpy()
})
```
## onnx graph with [netron](https://netron.app/)
![image](https://user-images.githubusercontent.com/28671653/111597397-139be880-8809-11eb-9e43-9a5fa47090ae.png)

## scripts with trtexec
```
trtexec --onnx=test_flatten.onnx --workspace=5 --verbose
```
## trace info
```
[03/18/2021-16:27:50] [V] [TRT] ModelImporter.cpp:103: Parsing node: Flatten_31 [Flatten]
[03/18/2021-16:27:50] [V] [TRT] ModelImporter.cpp:119: Searching for input: scores
[03/18/2021-16:27:50] [V] [TRT] ModelImporter.cpp:125: Flatten_31 [Flatten] inputs: [scores -> (2, 2, 10)], 
ERROR: onnx2trt_utils.cpp:188 In function convertAxis:
[8] Assertion failed: axis >= 0 && axis < nbDims
[03/18/2021-16:27:50] [E] Failed to parse onnx file
[03/18/2021-16:27:50] [E] Parsing model failed
[03/18/2021-16:27:50] [E] Engine creation failed
[03/18/2021-16:27:50] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec # trtexec --onnx=../../test/test_flatten.onnx --workspace=5 --verbose
```
## Environment

**TensorRT Version**:  TensorRT-7.2.1.6
**NVIDIA GPU**: GeForce GTX 1060
**NVIDIA Driver Version**: 455.45.01
**CUDA Version**: 10.2
**CUDNN Version**: 8.0
**Operating System**: ubuntu16.04
**Python Version (if applicable)**: 3.7.0
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.6.0
**Baremetal or Container (if so, version)**: 

Many thanks in advance.

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @SplitInfinity

**Comment by spandantiwari** [comment]
@RunningLeon - can you please try with the latest PyTorch version and see if this issue persists?

**Comment by RunningLeon** [comment]
@spandantiwari Hi, I could try with the latest PyTorch, but due to complex reasons, I'm afraid I have to solve this problem with PyTorch==1.6.0.

**Comment by RunningLeon** [comment]
@spandantiwari FYI, I have tested on PyTorch==1.6.0 and PyTorch==1.7.0 and this issue still happens.

**Comment by garymm** [comment]
According to the ONNX spec, I believe the graph is valid:

From [Flatten-11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Flatten-11): 

> The value for axis must be in the range [-r, r], where r is the rank of the input tensor.

So I think TensorRT may be buggy. You could try re-opening https://github.com/NVIDIA/TensorRT/issues/1134.
